coalib/settings/DocstringMetadata.py

```diff
@@ -24,54 +23,21 @@ def __init__(self, desc, param_dict, retval_desc):
     @classmethod
     def from_docstring(cls, docstring):
         """
-        Parses a python docstring. Usable attributes are:
-        :param
-        @param
-        :return
-        @return
+        Returns a dict after parsing the docstring.
         """
-        lines = inspect.cleandoc(docstring).split("\n")
-
-        parse_mode = cls._ParseMode.DESCRIPTION
-        cur_param = ""
-
-        desc = ""
+        parsed_doc = DocumentationComment(docstring).parse_documentation()
+        desc, retval_desc = "", ""
         param_dict = OrderedDict()
-        retval_desc = ""
-        for line in lines:
-            line = line.strip()
-
-            if line.startswith(":param ") or line.startswith("@param "):
-                parse_mode = cls._ParseMode.PARAM
-                splitted = line[7:].split(":", 1)
-                cur_param = splitted[0]
-                param_dict[cur_param] = splitted[1].strip()
-
-                continue
-
-            if line.startswith(":return: ") or line.startswith("@return: "):
-                parse_mode = cls._ParseMode.RETVAL
-                retval_desc = line[9:].strip()
-
-                continue
-
-            def concat_doc_parts(old: str, new: str):
-                if new != '' and not old.endswith('\n'):
-                    return old + ' ' + new
-
-                return old + (new if new != '' else '\n')
-
-            if parse_mode == cls._ParseMode.RETVAL:
-                retval_desc = concat_doc_parts(retval_desc, line)
-            elif parse_mode == cls._ParseMode.PARAM:
-                param_dict[cur_param] = concat_doc_parts(param_dict[cur_param],
-                                                         line)
-            else:
-                desc = concat_doc_parts(desc, line)
-
-        return (cls(desc=desc.strip(),
+        for part in parsed_doc:  # pragma: no cover
```

Comment:
Reason why this is not being covered:

By the current documentation parsing, there can be only one `Desc` section. When the loop passes through, it only finds it once and fails to go back to the for loop again. This causes a missed partial branch.

Need solutions :stuck_out_tongue_winking_eye: 

CC @Makman2